internal/CMakeLists.txt

```diff
@@ -1,22 +1,17 @@
 cmake_minimum_required(VERSION 3.19 FATAL_ERROR)
 
-# Setting multiple policies for additional packages
-# We set CMP0048 as a few packages don't set versions
-cmake_policy(SET CMP0048 NEW)
-set(CMAKE_POLICY_DEFAULT_CMP0048 NEW)
-cmake_policy(SET CMP0091 NEW)
-# For users of 3.24 and onwards, DOWNLOAD_TIMESTAMP will follow new behaviour
-# and use the current time instead of the archived timestamp (which can cause
-# issues with not rebuilding the content)
-if(POLICY CMP0135)
-  cmake_policy(SET CMP0135 NEW)
-  set(CMAKE_POLICY_DEFAULT_CMP0135 NEW)
-endif()
-# We set CMP0077 to prevent clearing of some of our variables
-cmake_policy(SET CMP0077 NEW)
-set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
 
-set(CMAKE_OSX_DEPLOYMENT_TARGET "10.15" CACHE STRING "Minimum macOS / OS X deployment version")
+#CMP0048 - a few packages don't set versions
+#CMP0077 - prevent clearing of certain variables
+#CMP0091 - Use CMAKE_MSVC_RUNTIME_LIBRARY Abstraction
+#CMP0135 - DOWNLOAD_TIMESTAMP follows new behaviour for CMake 3.24+
+foreach(policy CMP0048 CMP0077 CMP0091 CMP0135)
+  if(POLICY ${policy})
```

Comment:
What's this actually checking?

Reply:
This checks if the policy exists for the version of CMake that is configuring the project. If it does, it will set the policy's behaviour accordingly.